docs/OV_Runtime_UG/ov_infer_request.md

```diff
@@ -123,7 +122,10 @@ For more details please take a look too [Classification Sample Async](../../samp
 
 @endsphinxdirective
 
-You can use `ov::InferRequest::cancel()` method in case if you want to cancel the current inference request:
+> **NOTE**: Use weak reference of infer_request (`ov::InferRequest*`, `ov::InferRequest&`, `std::weal_ptr<ov::InferRequest>` and etc) in the callback, it is needed to avoid cyclic references.
+For more details please take a look too [Classification Sample Async](../../samples/cpp/classification_sample_async/README.md).
+
+You can use `ov::InferRequest::cancel` method in case if you want to abort execution of the current inference request:
```

Comment:
this works only on the CPU afaik... @vladimir-paramuzov ?

Reply:
but for other plugins it does nothing (no "not implemented" exception is thrown)